lib/ruby_lsp/requests/hover.rb

```diff
@@ -142,6 +150,43 @@ def generate_hover(name, location)
           contents: markdown_from_index_entries(name, entries),
         )
       end
+
+      sig { params(node: Prism::CallNode).void }
+      def generate_gem_hover(node)
+        first_argument = node.arguments&.arguments&.first
+        return unless first_argument.is_a?(Prism::StringNode)
+
+        spec = Gem::Specification.find_by_name(first_argument.content)
+        return unless spec
+
+        info = T.let(
+          [
+            spec.description,
+            spec.summary,
+            "This rubygem does not have a description or summary.",
+          ].find { |text| !text.nil? && !text.empty? },
+          String,
+        )
+
+        # Remove leading whitespace if a heredoc was used for the summary or description
+        info = info.gsub(/^ +/, "")
```

Comment:
Yeah, it is kind of smoothing over an issue upstream. I can see the merits of either. We could also move the version into a Code Lens (or do away with it entirely), and keep the hover content restricted to the plaintext summary/description context.

<img width="367" alt="Screenshot 2024-01-09 at 3 11 15 PM" src="https://github.com/Shopify/ruby-lsp/assets/16616777/087c3e34-5d2f-4a8c-a133-55be668a03e9">

Reply:
Hmm, even the example in the Rubygems docs uses `<<-`:

https://guides.rubygems.org/specification-reference/#description

Reply:
I can open a PR on https://github.com/rubygems/guides to use the squiggly heredoc. I would assume the docs pre-date the newer syntax itself?

Reply:
Yeah, that would be a helpful.

I still think we should strip the leading whitespace though, since some gems will never get around to updating, and it results in a disjointed experience for the user.

Reply:
Thinking about this more, even if it was supported in all non-EOL Ruby versions, it might actually restrict the gem from being installed on older versions due to invalid syntax, which might not be a reasonable trade-off.

Reply:
Opened https://github.com/rubygems/guides/pull/352 to address it in the RubyGems guides.

Reply:
> Thinking about this more, even if it was supported in all non-EOL Ruby versions, it might actually restrict the gem from being installed on older versions due to invalid syntax, which might not be a reasonable trade-off.

The `.gemspec` file is not actually part of the distributed `.gem` package so I don't think it would be a problem.